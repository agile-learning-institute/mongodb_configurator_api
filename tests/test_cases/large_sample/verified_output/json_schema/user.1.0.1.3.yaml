description: A user collection for testing the schema system
type: object
properties:
  _id:
    description: The unique identifier for a user
    type: string
    pattern: ^[0-9a-fA-F]{24}$
  user_name:
    description: Username
    type: string
    pattern: ^\S{1,40}$
  full_name:
    description: Users Full Name
    type: string
    pattern: ^[^\t\n\r]{0,255}$
  status:
    description: The status
    type: string
    enum:
    - draft
    - active
    - archived
  categories:
    description: A users list of categorized tags
    type: array
    items:
      description: A user category
      type: object
      properties:
        name:
          description: Category Name assigned by the user
          type: string
          pattern: ^\S{1,40}$
        category:
          description: The category type assigned by the user
          type: string
          enum:
          - work
          - personal
          - project
          - reference
        tags:
          description: A list of enumerated values assigned by the user
          type: array
          items:
            type: string
            enum:
            - urgent
            - important
            - normal
            - low
            - completed
            - in_progress
            - blocked
            - review
      required:
      - name
      - category
      - tags
      additionalProperties: false
  email:
    description: The person's email address
    type: string
    pattern: ^[^\s@]+@[^\s@]+\.[^\s@]+$
  phone:
    description: The person's phone number
    type: string
    pattern: ^\+1[2-9][0-9]{9}$
  last_saved:
    description: The last time this document was saved
    type: object
    properties:
      from_ip:
        description: Http Request remote IP address
        type: string
        pattern: ^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$
      by_user:
        description: ID Of User
        type: string
        pattern: ^\S{1,40}$
      at_time:
        description: The date-time when last updated
        type: string
        format: date-time
      correlation_id:
        description: The logging correlation ID of the update transaction
        type: string
        pattern: ^\S{1,40}$
    additionalProperties: false
    required:
    - from_ip
    - by_user
    - at_time
    - correlation_id
required:
- _id
- user_name
- status
- last_saved
additionalProperties: false
