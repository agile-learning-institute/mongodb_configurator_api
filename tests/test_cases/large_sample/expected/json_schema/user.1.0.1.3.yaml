title: User Collection
description: A user collection for testing the schema system
type: object
properties:
  _id:
    type: string
    description: The unique identifier for a user
    pattern: "^[0-9a-fA-F]{24}$"
  user_name:
    type: string
    description: Username
    pattern: "^[a-zA-Z0-9_-]{4,40}$"
  full_name:
    type: string
    description: Users Full Name
    pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
  status:
    type: string
    description: The status
    enum:
      - draft
      - active
      - archived
  categories:
    type: array
    description: A users list of categorized tags
    items:
      type: object
      description: A user category
      properties:
        name:
          type: string
          description: Category Name assigned by the user
          pattern: "^[a-zA-Z0-9_-]{4,40}$"
        category:
          type: string
          description: The category type assigned by the user
          enum:
            - work
            - personal
            - project
            - reference
        tags:
          type: array
          description: A list of enumerated values assigned by the user
          items:
            type: string
            enum:
              - urgent
              - important
              - normal
              - low
              - completed
              - in_progress
              - blocked
              - review
      required:
        - name
        - category
        - tags
      additionalProperties: false
  email:
    type: string
    description: The person's email address
    pattern: "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$"
  phone:
    type: string
    description: The person's phone number
    pattern: "^\\+1[2-9][0-9]{9}$"
  last_saved:
    title: Breadcrumb
    description: The last time this document was saved
    type: object
    properties:
      from_ip:
        description: Http Request remote IP address
        type: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      by_user:
        description: ID Of User
        type: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      at_time:
        description: The date-time when last updated
        type: string
        format: date-time
      correlation_id:
        description: The logging correlation ID of the update transaction
        type: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$" 
    required:
      - from_ip
      - by_user
      - at_time
      - correlation_id
    additionalProperties: false
required:
  - _id
  - user_name
  - status
  - last_saved
additionalProperties: false